src/fetchers/onchain.js

```diff
@@ -0,0 +1,199 @@
+import web3 from "web3";
+import fetch from "node-fetch";
+import { parse } from "../parsers/onchain";
+
+const FETCH_TIMEOUT = 30000;
+
+let Web3Local = new web3(web3.givenProvider || process.env.MAINNET_RPC_URL);
+
+let Web3Instances = [];
+let Web3InstanceMax = 10;
+for (let i = 0; i < Web3InstanceMax; i++) {
+  Web3Instances.push(new web3(process.env.MAINNET_RPC_URL));
+}
+
+const getWeb3 = () => {
+  const instance = Web3Instances.shift();
+  Web3Instances.push(instance);
+  return instance;
+};
+
+setInterval(() => {
```

Comment:
When we move the metadata service, this should be a cron job rather than in memory logic